modules/skybrowser/src/browser.cpp

```diff
@@ -0,0 +1,219 @@
+        std::optional<bool> reload;
+    };
+
+#include "browser_codegen.cpp"
+
+} // namespace
+
+namespace openspace {
+
+void Browser::RenderHandler::draw() {}
+
+void Browser::RenderHandler::render() {}
+
+void Browser::RenderHandler::setTexture(GLuint t) {
+    _texture = t;
+}
+
+Browser::Browser(const ghoul::Dictionary& dictionary)
+    : _url(UrlInfo)
+    , _browserPixeldimensions(DimensionsInfo, glm::vec2(500.f), glm::vec2(10.f), glm::vec2(3000.f))
+    , _reload(ReloadInfo)
+{ 
+    if (dictionary.hasValue<std::string>(UrlInfo.identifier)) {
+        _url = dictionary.value<std::string>(UrlInfo.identifier);
+    }
+    // Handle target dimension property
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+    _url = p.url.value_or(_url);
+    _browserPixeldimensions = p.dimensions.value_or(_browserPixeldimensions);
+
+    glm::vec2 windowDimensions = global::windowDelegate->currentSubwindowSize();
+    _browserPixeldimensions = windowDimensions;
+
+    _url.onChange([this]() { _isUrlDirty = true; });
+    _browserPixeldimensions.onChange([this]() { _isDimensionsDirty = true; });
+    _reload.onChange([this]() { _shouldReload = true; });
+
+    // Create browser and render handler
+    _renderHandler = new RenderHandler();
+    _keyboardHandler = new WebKeyboardHandler();
+    _browserInstance = std::make_unique<BrowserInstance>(
+        _renderHandler,
+        _keyboardHandler
+    );
+
+    WebBrowserModule* webBrowser = global::moduleEngine->module<WebBrowserModule>();
+    if (webBrowser) {
+        webBrowser->addBrowser(_browserInstance.get());
+    } 
+}
+
+Browser::~Browser() {}
+
+bool Browser::initializeGL() {
+    _texture = std::make_unique<ghoul::opengl::Texture>(
+        glm::uvec3(glm::ivec2(_browserPixeldimensions.value()), 1),
+        GL_TEXTURE_2D
+    );
+       
+    _renderHandler->setTexture(*_texture);
+
+    _browserInstance->initialize();
+    _browserInstance->loadUrl(_url);
+    return isReady();
+}
+
+bool Browser::deinitializeGL() {
+    _renderHandler->setTexture(0);
+
+    _texture = nullptr;
+
+    LDEBUG(fmt::format("Deinitializing browser: {}", _url.value()));
+
+    _browserInstance->close(true);
+
+    WebBrowserModule* webBrowser = global::moduleEngine->module<WebBrowserModule>();
+    if (webBrowser) {
+        webBrowser->removeBrowser(_browserInstance.get());
+        _browserInstance.reset();
+    }
+    else {
+        LWARNING("Could not find WebBrowserModule");
+    }
+
+    return true;
+}
+
+void Browser::render() {
+    if (!_renderHandler->isTextureReady()) {
+        return;
+    }
+    _renderHandler->updateTexture();
+}
+
+void Browser::update() {
+    if (_isUrlDirty) {
+        _browserInstance->loadUrl(_url);
+        _isUrlDirty = false;
+    }
+    if (_isDimensionsDirty) {
+        if (_browserPixeldimensions.value().x > 0 &&
+            _browserPixeldimensions.value().y > 0) 
+        {
+            _browserInstance->reshape(_browserPixeldimensions.value());
```

Comment:
Shouldn't have to use `value()` here_
`_browserInstance->reshape(_browserPixeldimensions);`